go-crud/books/controllers.go

```diff
@@ -0,0 +1,257 @@
+package books
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var data []book
+
+func loadBookFile() error {
+	_, err := os.Stat("books.json")
+
+	if os.IsNotExist(err) {
+		str, err := json.Marshal(data)
+
+		if err != nil {
+			log.Println(err)
+			return err
+		}
+		err = ioutil.WriteFile("books.json", str, 0644)
+		if err != nil {
+			log.Println(err)
+			return err
+		}
+	}
+
+	rawData, err := ioutil.ReadFile("books.json")
+
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+
+	// fmt.Print("data:  ", string(rawData))
+	err = json.Unmarshal(rawData, &data)
+	if string(rawData) == "null" {
+		data = make([]book, 0)
+	}
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+
+	return nil
+}
+
+func createBook(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "POST" {
+		err := loadBookFile()
+
+		if err != nil {
+			log.Println(err.Error())
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		var b book
+
+		err = json.NewDecoder(r.Body).Decode(&b)
+		if err != nil {
+			if err == io.EOF {
+				http.Error(w, "Empty Request Body", http.StatusBadRequest)
+				return
+			}
+			log.Println(err.Error())
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		_, err = validateBook(b)
+
+		if err != nil {
+			log.Println(err.Error())
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		if len(data) == 0 {
+			b.Id = 1
+		} else {
+			b.Id = data[len(data)-1].Id + 1
+		}
+
+		writeFile(data, b)
```

Comment:
This method omits errors this is not good. Let it return errors and handle it.
Also you are using this custom method and `ioutil.WriteFile` maybe we should decide on only one.